### ISSUE TITLE
mne_bids.copyfiles.copyfile_eeglab assumes setname (from EEG struct) is equal to filename. 

### ISSUE THREAD

**Issue by kaare-mikkelsen** [mne-tools/mne-bids#1120]
### Description of the problem

Offending command:

mb.write_raw_bids(raw,bids_path,overwrite=True,verbose=True)

Both 'raw' and bids_path are describing real files which exist. However, I get an error from shutil, saying that the .fdt file does not exist. This turns out to be caused by the definition of 'fdt_name' in copyfile_eeglab, which incorrectly assumes that the fdt_name should be based on the EEG.setname and not the actual filename of the .set file. It is the custom in EEGLAB to have matching .set and .fdt files, but these are usually named independently of the setname. 

### Steps to reproduce

```Python
try to move .set file with mismatch between setname and filename.
```

### Expected results

no error, setname does not dictate filenames. 

### Actual results

files have to be named according to setname, which is needlessly rigid

### Additional information

mb.__version__: 0.11.1

**Comment by sappelhoff** [comment]
@kaare-mikkelsen do you want to submit a PR that will allow mne-bids to deal with all previous cases as well as the case you are bringing up right here?

**Comment by kaare-mikkelsen** [comment]
@sappelhoff I would think that all previous cases would also be dealt with if fdt_name matched the name of the .set file and not the 'setname' inside the .set file. EEGLAB will always save the datasets such that the names of the .set and .fdt files are identical. whether those names match the setname is not a given though. 

but, are you simply asking me to make the change and submit the PR? I might be misunderstanding you :)

**Comment by hoechenberger** [comment]
> whether those names match the setname is not a given though.

Are you absolutely sure about this? I'm wondering if the files you're dealing with have simply been renamed without updating the setname value.

In any case it would be good to stick with our existing approach as a first pass and if it fails, fall back to something like `Path(set_fname).with_suffix('.fdt'`

**Comment by kaare-mikkelsen** [comment]
no, I think the difference here is that I am working with files that have been saved by calling 'pop_saveset' directly in matlab. That function takes as optional input a filename (which I guess has been given in this case). This filename is used for all files generated (either one or two). if the optional input is not given, the function uses the 'filename' in the EEG-struct (note that this is _also_ not the setname). It is often quite reasonable to have the setname match the filename, or to pass the setname to the pop_saveset function, but that is not enforced by eeglab.
 I agree that using the filename of the .set file as a fallback is a pragmatic solution, but I am unsure what cases it is meant to support? saveset only takes a single file name as input, so the .set and .fdt files will _always_ have the same name (unless someone is manually renaming one of them, which seems like terrible practice, and would mess up the current practice too). And copyfile_eeglab already knows the name of the .set-file, otherwise it would be failing much earlier than what is happening here.

**Comment by sappelhoff** [comment]
Sounds good, please send over a PR then!

**Comment by sappelhoff** [comment]
@laemtl you are working with EEGLAB files -- do you perhaps have an opinion about this issue?

**Comment by sappelhoff** [comment]
> This turns out to be caused by the definition of 'fdt_name' in copyfile_eeglab, which incorrectly assumes that the fdt_name should be based on the EEG.setname and not the actual filename of the .set file. 

@kaare-mikkelsen 

could you please link to the lines in the code to directly point to where you think the problem lies?

**Comment by kaare-mikkelsen** [comment]
this line: 

`fdt_fname = eeg['data'][0, 0][0]`

Immediately after the line 'if has_fdt_link:' in https://mne.tools/mne-bids/dev/_modules/mne_bids/copyfiles.html#copyfile_eeglab. 

As far as I can see, it collects the value from EEG.setname. In EEGLAB, the 'data' field is just a matrix. I don't know how the eeg['data'] field in python is populated, I just observe that fdt_fname = eeg['data'][0, 0][0] returns a value which, in my concrete example, is only present as the setname in the EEG struct.

**Comment by sappelhoff** [comment]
> In EEGLAB, the 'data' field is just a matrix

As far as I know, the `data` field may be:

1. a matrix with actual values ... in that case, the full EEG data consists only of a single `.set` file
2. a string pointing to a `.fdt` file ... in that case, the full EEG data consists of two files: the `.set` file and the `.fdt` file

Would you mind sharing an example file that produces the error for you? Or a minimal working example in matlab to produce a set+fdt file that elicits this error?

I think we'd need to investigate this more closely to progress.

**Comment by kaare-mikkelsen** [comment]
hmm, it seems there might something odd about how the .set file has been created. I can reproduce my error, but if I load the set in matlab, reduce the data matrix and resave it, the problem disappears. I will try to get hold of the specific script that created these files (it seems to be an issue with all of them), and see if I can understand what is going on, or upload a small working example here.

**Comment by sappelhoff** [comment]
>  I will try to get hold of the specific script that created these files (it seems to be an issue with all of them), and see if I can understand what is going on, or upload a small working example here.

any news on this front @kaare-mikkelsen ?

**Comment by kaare-mikkelsen** [comment]
Hi @sappelhoff 
Yes, I found the script, but it was very specialized, and I was not convinced that whatever was happening there truly generalized to other situations. I still think that assuming that setname and filename is weird, but will close this issue for now until I find a more fitting example of the issue. Thanks for reminding me to close this :)
